### PR TITLE
Resolving unhandled promise rejection errors, #229

### PR DESCRIPTION
**Pull request by rorticus**
> The tests, they did throw uncaught promises a lot.
> Dispatching their promises, the results were forgot.
> Merge this PR,
> and then you will see,
> The test logs don't error, they're clean as can be.

Also, we'll probably want someone familiar with the streams to give this a quick glance to confirm that I'm not hiding any _important_ errors. There shouldn't be anything caught now that wasn't before, but, you never know!

### CODE REVIEW

File: src/request/node.ts [hunk 1]
```diff
@@ -282,7 +282,8 @@ export default function node<T>(url: string, options: NodeRequestOptions<T> = {}
 						},
 						function (error: RequestError<T>) {
 							if (options.streamTarget) {
-								options.streamTarget.abort(error);
+								options.streamTarget.abort(error).catch(() => {
```
Comment (kitsonk) [inline src/request/node.ts:287]: It would be good to add a comment on this, because they look "odd" otherwise.

File: tests/unit/request/node.ts [hunk 2]
```diff
@@ -122,6 +122,12 @@ const responseData: { [url: string]: DummyResponse } = {
 	}
 };
 
+process.on('unhandledRejection', (reason: any, promise: any) => {
```
Comment (kitsonk) [inline tests/unit/request/node.ts]: Is this still desired/intended?  Seems like it was what was used to chase down the issues.

**Review by kitsonk — COMMENTED** [review]
Just a couple of comments

File: tests/unit/request/node.ts (hunk 2 above)
Comment (rorticus) [reply tests/unit/request/node.ts]: Good catch, you're right, that shouldn't be there! :)

File: src/request/node.ts (hunk 1 above)
Comment (rorticus) [reply src/request/node.ts:287]: Do you think I should comment on all the places I did this, or just this one?

**Review by kitsonk — APPROVED** [review]
👍

**Pull request merged**